staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -158,6 +177,31 @@ func (q *delayingType) ShutDown() {
 	})
 }
 
+// Add overrides the Add() method provided by Interface so that we properly consider the
+// state of the delaying queue when performing an immediate add.
```

Comment:
I'd write it more like this:

```
// Add immediately places `item` in the queue for processing, and removes it from the
// waiting/delaying queue if it is already waiting. This effectively ignores any delay previously
// passed to AddAfter with this item.
```

Reply:
Yes that is much better! :) Going with your suggestion.